### ISSUE TITLE
Ubuntu 16.10 "Yakkety" instructions for engine/installation/linux/ubuntulinux.md

### ISSUE THREAD

**Issue by preston** [docker/docker.github.io#329]
URL: [https://docs.docker.com/engine/installation/linux/ubuntulinux.md](https://docs.docker.com/engine/installation/linux/ubuntulinux/)

Ubuntu 16.10 was recently released, and I'm sure a lot of people will be following an officially-supported upgrade path from 16.04 (LTS) to the non-LTS 16.10 "yakkety" release. The Docker instructions do not currently make it clear how -- if it all -- this can be done safely.

Also note that the referenced apt repo at https://apt.dockerproject.org/repo/dists/ does not seem to currently offer a build specifically for yakkety.

**Comment by konstruktoid** [comment]
There are currently no .deb build for yakkety in stable nor in experimental.

**Comment by konstruktoid** [comment]
See https://github.com/docker/docker/issues/27521

**Comment by mstanleyjones** [comment]
Ubuntu 1.12x has no builds for Yakkety. See https://github.com/docker/docker/issues/27521#issuecomment-260669710 for information about trying the 1.13x testing builds on Yakkety. Thanks!

**Comment by jhwheeler** [comment]
So what's the new situation with Yakkety? Following the [docs for Ubuntu install](https://docs.docker.com/engine/installation/linux/ubuntu/#install-using-the-repository), I've tried with:

`sudo add-apt-repository \
   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
   $(lsb_release -cs) \
   stable"`

Yet to no avail. 

I get the following: 

`W: The repository 'https://download.docker.com/linux/ubuntu (lsb_release Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: The repository 'https://download.docker.com/linux/ubuntu\ (lsb_release Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch https://download.docker.com/linux/ubuntu/dists/(lsb_release/-cs)/binary-amd64/Packages  404  Not Found
E: Failed to fetch https://download.docker.com/linux/ubuntu\/dists/(lsb_release/-cs)/binary-amd64/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.`

**Comment by mstanleyjones** [comment]
Yakkety is not supported. You can see the supported versions at the top of the topic. You can still use static binaries. :)

**Comment by andrewhsu** [comment]
@jhwheeler from the output of the `add-apt-repository` command you showed, it looks like the execution of the command was escaped so that some of the special characters like the `$` were not evaluated by bash.

From the [documentation](https://docs.docker.com/engine/installation/linux/ubuntu/#install-using-the-repository) the command you want to run is:
```
$ sudo add-apt-repository \
   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
   $(lsb_release -cs) \
   stable"
```

Where the `$(lsb_release -cs)` should evaluate to `yakkety` instead of ending up like `(lsb_release ` like what you had in the output.

If executed correctly when you update the repository, the output should look like this:
```
$ apt-get update
Hit:1 http://archive.ubuntu.com/ubuntu yakkety InRelease
Get:2 http://security.ubuntu.com/ubuntu yakkety-security InRelease [102 kB]
Get:3 http://archive.ubuntu.com/ubuntu yakkety-updates InRelease [102 kB]          
Get:4 https://download.docker.com/linux/ubuntu yakkety InRelease [20.1 kB]
Get:5 https://download.docker.com/linux/ubuntu yakkety/stable amd64 Packages [1235 B]            
Get:6 http://archive.ubuntu.com/ubuntu yakkety-backports InRelease [102 kB]                
Fetched 328 kB in 1s (169 kB/s)    
Reading package lists... Done
```

And you can then proceed to `$ sudo apt-get install docker-ce`.

**Comment by jhwheeler** [comment]
@andrewhsu Yes, that's what I thought as well; it seems the `$(lsb_release -cs)` isn't being interpreted as `yakkety`. Not sure why it's escaping incorrectly; I've tried both copy-pasting this in directly and typing it by hand, both by leaving it on multiple lines as it is above and putting all on one line. I've also tried replacing `$(lsb_release -cs)` with `yakkety` directly. None of these seemed to work. In any regard, in the end I just tried `sudo apt-get install docker-ce` and now it's working perfectly. I suppose I'll have to manage the updates on my own this way?
